### ISSUE TITLE
MathJax

### ISSUE THREAD

**Issue by danielballan** [mpld3/mpld3#14]
Thanks to @jkfurtney we have basic support for log axes. But the tick labels can get ugly. The underlying tick labels contain TeX: `Text(10,0,'$\\mathdefault{10^{1}}$')`. This seems like a good excuse to see how difficult it would be to render tick labels using MathJax. Ultimately, of course, it would be nice to support MathJax for all text.
- [MathJax-free approach using CSS](http://bl.ocks.org/mbostock/6738229) -- useful reference whether or not MathJax is part of our  solution
- [rendering MathJax as a `foreignObject` inside an svg](https://github.com/mathjax/MathJax/issues/394), specifically [this comment](https://github.com/mathjax/MathJax/issues/394#issuecomment-13566519)
- [not-so-helpful but related SO post](http://stackoverflow.com/questions/14654748/how-to-render-a-svg-inserted-by-mathjax-onto-a-canvas)

UPDATE: OK, we don't have to reinvent the wheel of d3 + mathjax. Roundup:
- https://gist.github.com/chrisbrich/3391642

Jake, I'd be especially interested in your reaction to the CSS approach (first link) before I dive too far into this.

**Comment by ichuang** [comment]
Inspired by @aflaxman, here's a working example showing mathjax rendering of math in svg output from mpld3:

   http://jsfiddle.net/ichuang/0ozb4zy5/

![image](https://cloud.githubusercontent.com/assets/1590761/5600531/953af554-92a9-11e4-94d4-a98ee7ddf5aa.png)


It uses a hacked up version of Jason Sachs' code:

   https://github.com/ichuang/svg_mathjax2

Note that the mathjax rendering logic is different - I found it better to directly queue rendering requests, rather than adding a hook into MathJax's begin and end typeset events.  There are also many debugging lines.

The output text positions and clip box need adjustment...

**Comment by aflaxman** [comment]
So cool!  We are getting close.  I wish it used $$ notation...  is that an easy tweak?

**Comment by danielballan** [comment]
@ichuang Very cool. Would you be willing to work this into a PR, or would you like help doing that?

**Comment by ichuang** [comment]
help appreciated.  I don't have enough time to do it myself right now.

**Comment by avipersin** [comment]
I came across this issue the other day and have implemented @ichuang 's solution successfully.  A couple things I thought I would mention:

1. Matplotlib accepts a larger set of Identifiers than does the regex used in this solution.  I had to change all of my Matplotlib Tex strings to comply with the findSVGMathJax() function.

2. Instead of adding   ``` $(document).ready(function() {new Svg_MathJax().install();});``` to each SVG instance, I just added ```window.addEventListener('load', function () {new Svg_MathJax().install();});``` to the entire page.  And for asynchronous SVG's, I've utilized the callback function. 

Thanks for the solution, it makes this library so much more powerful.

**Comment by amichaut** [comment]
Is it planned to add latex rendering to the next release?

**Comment by Vital-Fernandez** [comment]
Hi! I am using mpld3 to have interactive plots in streamlit. It works very well except for the latex rendering.  Is it actually supported?

**Comment by vladh** [comment]
@Vital-Fernandez I don't think this was ever implemented, PR welcome!
